sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonAdapter.java

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.jackson;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.implementation.AccessibleByteArrayOutputStream;
+import com.azure.core.serializer.json.jackson.implementation.ObjectMapperShim;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.DateTimeRfc1123;
+import com.azure.core.util.ExpandableStringEnum;
+import com.azure.core.util.Header;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.CollectionFormat;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Type;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Implementation of {@link SerializerAdapter} that uses Jackson.
+ * <p>
+ * This is similar to {@code JacksonAdapter} found in the {@code com.azure:azure-core} package and will serve as the
+ * long term replacement plan for the {@code JacksonAdapter} found in {@code com.azure:azure-core}.
+ */
+public final class JacksonAdapter implements SerializerAdapter {
+    private static final ClientLogger LOGGER = new ClientLogger(JacksonAdapter.class);
+
+    // Enum Singleton Pattern
+    private enum GlobalXmlMapper {
+        XML_MAPPER(ObjectMapperShim.createXmlMapper());
```

Comment:
wonder if we need xml here? Could we special-case storage somehow and switch servicebus admin to stream -tyle so that XML is not needed here?

Reply:
Made it so XML serialization is only supported when Jackson Dataformat XML is on the classpath and it's not provided by the SDKs (or at least this library).